R/resample_agb.R

```diff
@@ -93,6 +93,15 @@ resample_agb <- function(genus,
     list_agb <-
       eval(parse(text = new_equation)) * dfsub$output_units_cf
   } else {
+    # FIXME: The iteration in this 'else' branch appears to be wrong, likely
+    # because lapply() syntax is mixed with for() syntax. Operations over all
+    # items in a vector appear to be mixed with operations over a single element
+    # of a vector. The result is that the length of `list_dbh` is different from
+    # the length of `list_agb`. Please revise this chunk and write a test to
+    # ensure the result is what you expect and to avoid this bug from comming
+    # back in the future. As an intermediate step to fix this bug it may be
+    # easiest to extract the logic of this chunk in a little helper function
+    # then test that helper in isolation.
```

Comment:
Let me know if this is unclear.

Reply:
Hi Mauro, 
Thanks for the explanation, but I'm not sure I understand the problem you mention. 
The previous logic was that the `sampled_dbh` variable was used by calling the `new_equation` variable, which is just a character string. For example, `new_equation` is `"1.2927*((sampled_dbh*0.393701)^2)^1.36723"` which is then evaluated at L. 101, thus calling the `sampled_dbh` variable to calculate `agb`. 

With the suggested changes, this no longer works because `new_equation` is not a character string representing the equation to be evaluated.

So I suggest reverting the code changes because the main reason for the changes was that `sampled_dbh` was not used, which is not exactly true (although it is not detected because it is used inside the `eval()` function). If there is a problem with my previous logic, or if the explanation was unclear, we can discuss it further and perhaps think of another way to do it.

Reply:
My bad! Sorry for not fully understanding and adding confusion. Go ahead and revert that change :-)

As for that variable not being used — there may be a hack to keep R CMD check happy. I’ll try a few things and will let you know what I find; but I think your explanation will do for now.